src/containers/TemplateBuilder/shared/TextIO.tsx

```diff
@@ -6,8 +6,8 @@ const iconLink = 'https://react.semantic-ui.com/elements/icon/'
 const colourLink = 'https://htmlcolorcodes.com/'
 
 type TextIOprops = {
-  text?: string
-  title?: string
+  text?: string | null
+  title?: string | null
```

Comment:
Was causing problems setting the "code" to empty string, as I've added database constraint to make the code unique per template, and this was making them all empty strings if not specified, which broke the constraint. 

So now they can remain null if they start as null

Reply:
Had to revisit this, as it caused another bug #1243. 

I've made changes now to allow the text values to be `null` throughout, and where it *can't* be, it falls back to an empty string. Not super happy with it, but best I could do without a significant re-factor.

a9802b10